### PR TITLE
[14.0] Migrating improvements

### PR DESCRIPTION
**Pull request by etobella**
Necessary commits for 14.0 `edi` version

Cherry-picked from 13.0

@simahawk

### CODE REVIEW

File: component/core.py [hunk 1]
```diff
@@ -346,14 +346,58 @@ def component_by_name(self, name, model_name=None):
...
             self.collection._name, usage=usage, model_name=model_name
         )
+        matching_components = []
+        for cls in component_classes:
+            try:
+                matching = cls._component_match(
+                    self, usage=usage, model_name=model_name, **kw
+                )
+            except TypeError as err:
+                # Backward compat
```
Comment (simahawk) [inline component/core.py:360]: I think we can drop this

Comment (etobella) [reply component/core.py:360]: I would agree with you if this was the migration of the module, but component was already on 14, could be implementations with the old behavior, isn't it?

Comment (simahawk) [reply component/core.py:360]: oh yeah... well, it's unlikely that someone apart from @guewen customized it  but... ok, let's stay safe and do it on v15 :+1:

**Review by simahawk — APPROVED** [review]

**Pull request merged**